### PR TITLE
chore: remove method PrivateKeyFromHex

### LINKED ISSUE #451: remove secp256k1.PrivateKeyFromHex method
**Is your feature request related to a problem? Please describe.**
This method should be removed as it is not the best way to get a private key.

**Describe the solution you'd like**
- [ ] Remove method
- [ ] Replace with `PrivateKeyFromBytes`

### PR DESCRIPTION
**Pull request by Kifen**
fixes #451

### CODE REVIEW

File: crypto/secp256k1/private_test.go [hunk 1]
```diff
@@ -128,7 +128,7 @@ func TestPrivateKeyFromHex(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := PrivateKeyFromHex(tt.args.hexkey)
+			got, err := PrivateKeyFromBytes([]byte(tt.args.hexkey))
```
Comment (robdefeo) [inline crypto/secp256k1/private_test.go]: I think its best to remove this test as it is not needed. Please remove `func TestPrivateKeyFromHex` as `PrivateKeyFromBytes` is already tested elsewhere

Comment (Kifen) [reply crypto/secp256k1/private_test.go]: will do

Comment (Kifen) [reply crypto/secp256k1/private_test.go]: Removed `TestPrivateKeyFromHex` from secp256k1/private_test.go

File: crypto/multikey/kind_test.go [hunk 2]
```diff
@@ -86,7 +86,7 @@ func TestKeyKindFromSignature(t *testing.T) {
 			"success-secp256k1-sofia-key",
 			args{
 				func() []byte {
-					k, _ := secp256k1.PrivateKeyFromHex("01901E63389EF02EAA7C5782E08B40D98FAEF835F28BD144EECF5614A415943F")
+					k, _ := secp256k1.PrivateKeyFromBytes([]byte("01901E63389EF02EAA7C5782E08B40D98FAEF835F28BD144EECF5614A415943F"))
```
Comment (robdefeo) [inline crypto/multikey/kind_test.go]: `[]byte(string)` will attempt a literal conversation not work as expected as the `[]byte` is encoded as a hex string. For testing you can use the `encodingtest.MustDecodeHex` e.g. replace `[]byte` with `encodingtest.MustDecodeHex`.

File: crypto/multikey/kind_test.go [hunk 3]
```diff
@@ -117,7 +117,7 @@ func TestKeyKindFromSignature(t *testing.T) {
 			"success-secp256k1-charlotte-key",
 			args{
 				func() []byte {
-					k, _ := secp256k1.PrivateKeyFromHex("DF4BA9F6106AD2846472F759476535E55C5805D8337DF5A11C3B139F438B98B3")
+					k, _ := secp256k1.PrivateKeyFromBytes([]byte("DF4BA9F6106AD2846472F759476535E55C5805D8337DF5A11C3B139F438B98B3"))
```
Comment (robdefeo) [inline crypto/multikey/kind_test.go]: `[]byte(string)` will attempt a literal conversation not work as expected as the `[]byte` is encoded as a hex string. For testing you can use the `encodingtest.MustDecodeHex` e.g. replace `[]byte` with `encodingtest.MustDecodeHex`.

File: crypto/multikey/kind_test.go [hunk 4]
```diff
@@ -148,7 +148,7 @@ func TestKeyKindFromSignature(t *testing.T) {
 			"success-secp256k1-mix-key-kinds",
 			args{
 				func() []byte {
-					k, _ := secp256k1.PrivateKeyFromHex("DF4BA9F6106AD2846472F759476535E55C5805D8337DF5A11C3B139F438B98B3")
+					k, _ := secp256k1.PrivateKeyFromBytes([]byte("DF4BA9F6106AD2846472F759476535E55C5805D8337DF5A11C3B139F438B98B3"))
```
Comment (robdefeo) [inline crypto/multikey/kind_test.go]: `[]byte(string)` will attempt a literal conversation not work as expected as the `[]byte` is encoded as a hex string. For testing you can use the `encodingtest.MustDecodeHex` e.g. replace `[]byte` with `encodingtest.MustDecodeHex`.

File: crypto/multikey/kind_test.go [hunk 5]
```diff
@@ -176,7 +176,7 @@ func TestKeyKindFromSignature(t *testing.T) {
 			"err-no-key-kinds",
 			args{
 				func() []byte {
-					k, _ := secp256k1.PrivateKeyFromHex("DF4BA9F6106AD2846472F759476535E55C5805D8337DF5A11C3B139F438B98B3")
+					k, _ := secp256k1.PrivateKeyFromBytes([]byte("DF4BA9F6106AD2846472F759476535E55C5805D8337DF5A11C3B139F438B98B3"))
```
Comment (robdefeo) [inline crypto/multikey/kind_test.go]: `[]byte(string)` will attempt a literal conversation not work as expected as the `[]byte` is encoded as a hex string. For testing you can use the `encodingtest.MustDecodeHex` e.g. replace `[]byte` with `encodingtest.MustDecodeHex`.

**Pull request merged**